src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -661,20 +649,19 @@ def delete_connectedk8s(cmd, client, resource_group_name, cluster_name,
         arm_hash = hashlib.sha256(armid.lower().encode('utf-8')).hexdigest()
 
         if check_proxy_kubeconfig(kube_config, kube_context, arm_hash):
-            telemetry.set_user_fault()
             telemetry.set_exception(exception='Encountered proxy kubeconfig during deletion.', fault_type=consts.Proxy_Kubeconfig_During_Deletion_Fault_Type,
                                     summary='The resource cannot be deleted as user is using proxy kubeconfig.')
-            raise CLIError("az connectedk8s delete is not supported when using the Cluster Connect kubeconfig. Run the az connectedk8s delete command with your kubeconfig file pointing to the actual Kubernetes cluster to ensure that the agents are cleaned up successfully as part of the delete command.")
+            raise ClientRequestError("az connectedk8s delete is not supported when using the Cluster Connect kubeconfig.", recommendation="Run the az connectedk8s delete command with your kubeconfig file pointing to the actual Kubernetes cluster to ensure that the agents are cleaned up successfully as part of the delete command.")
 
         delete_cc_resource(client, resource_group_name, cluster_name, no_wait)
     else:
         telemetry.set_user_fault()
```

Comment:
same as above, we can remove `set_user_fault()` once raising the new error type.

Reply:
Ack.